config/config.go

```diff
@@ -0,0 +1,199 @@
+package config
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/giantswarm/gsclientgen"
+)
+
+// struct used for YAML serialization of settings
+type configStruct struct {
+	Token        string
+	Email        string
+	Updated      string
+	Organization string
+	Cluster      string
+}
+
+const (
+	// ConfigFileName is the name of the configuration file
+	ConfigFileName string = "config.yaml"
+
+	// ProgramName is the name of this program
+	ProgramName string = "gsctl"
+)
+
+var (
+	// Version is the version number, to be set on build by the go linker
+	Version string
+
+	// BuildDate is the build date, to be set on build by the go linker
+	BuildDate string
+
+	// Commit is the latest git commit hash, to be set on build by the go linker
+	Commit string
+
+	// Config holds the configuration variables
+	Config *configStruct
+
+	// ConfigDirPath is the path of the directory holding our config file
+	ConfigDirPath string
+
+	// ConfigFilePath is the path of the configuration file
+	ConfigFilePath string
+
+	// KubeConfigPaths is the path(s) of kubeconfig files as slice of strings
+	KubeConfigPaths []string
+
+	// SystemUser is the current system user as user.User (os/user)
+	SystemUser *user.User
+)
+
+func init() {
+
+	SystemUser, userErr := user.Current()
+	checkErr(userErr)
+
+	ConfigDirPath = SystemUser.HomeDir + string(os.PathSeparator) + "." + ProgramName
+	ConfigFilePath = ConfigDirPath + string(os.PathSeparator) + ConfigFileName
+	KubeConfigPaths, _ = getKubeconfigPaths(SystemUser.HomeDir)
+
+	myConfig, err := readFromFile(ConfigFilePath)
+	checkErr(err)
+	Config = myConfig
+}
+
+func checkErr(err error) {
+	if err != nil {
+		panic(err.Error())
+	}
+}
+
+// WriteToFile writes the configuration data to a YAML file
+func WriteToFile() error {
+
+	// ensure directory
+	os.MkdirAll(ConfigDirPath, 0700)
+
+	// last modified date
+	Config.Updated = time.Now().Format(time.RFC3339)
+
+	yamlBytes, yamlErr := yaml.Marshal(&Config)
+	if yamlErr != nil {
+		return yamlErr
+	}
+
+	writeErr := ioutil.WriteFile(ConfigFilePath, yamlBytes, 0600)
+	if writeErr != nil {
+		return writeErr
+	}
+
+	return nil
+}
+
+// ReadFromFile reads configuration from the YAML config file
+func readFromFile(filePath string) (*configStruct, error) {
+	myConfig := new(configStruct)
+
+	data, readErr := ioutil.ReadFile(filePath)
+	if readErr != nil {
+		if os.IsNotExist(readErr) {
+			// ignore if file does not exist,
+			// as this is not an error.
+			return myConfig, nil
+		}
+		return nil, readErr
+	}
+
+	yamlErr := yaml.Unmarshal(data, &myConfig)
+	if yamlErr != nil {
+		return nil, yamlErr
+	}
+
+	return myConfig, nil
+}
+
+// getKubeconfigPaths returns a slice of paths to known kubeconfig files.
+func getKubeconfigPaths(homeDir string) ([]string, error) {
```

Comment:
Your return signature is `([]string, error)`, but I can't actually see any errors being returned - can we simplify to just returning `[]string`?